src/overseer/core.clj

```diff
@@ -85,6 +85,27 @@
   []
   (quot (System/currentTimeMillis) 1000))
 
+; Core functions encompassing the system's interaction with storage backends
+; These functions effectively represent the finite state machine transitions
+; that a job goes through:
+;
+;      unstarted ---------+
+;       |  ^              |
+;       |  |              |
+;       v  |              |
+;      started -----------+
+;      /     \            |
+;     /        \          |
+; finished   failed   aborted
+;
+; The system internally handles the transitions:
+;   (unstarted -> started)
+;   (started -> finished)
+;   (started -> failed)
+;
```

Comment:
Well the * -> aborted case is documented (requested by caller) and the dependent stuff is sort of implied by the system's definition of abort. Regardless, I'll make that aspect more clear